src/screens/QuestionScreen.tsx

```diff
@@ -0,0 +1,272 @@
+import React, { useState } from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  ScrollView,
+  Alert,
+} from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { useTaskContext } from '../context/TaskContext';
+import { TaskScore } from '../types';
+
+const QuestionScreen: React.FC = () => {
+  const navigation = useNavigation();
+  const { questions, tasks, currentScores, updateTaskScore, addAnsweredQuestion, resetScores } = useTaskContext();
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+
+  const currentQuestion = questions[currentQuestionIndex];
+
+  const handleAnswer = (answer: boolean) => {
+    if (!currentQuestion) return;
+
+    // Store the answer in context
+    addAnsweredQuestion(currentQuestion.id, answer);
+
+    // Update task scores based on the answer
+    const property = currentQuestion.property;
+    const response = answer;
+    
+    tasks.forEach(task => {
+      if (task.properties[property] === response) {
+        const currentScore = currentScores.find((s: TaskScore) => s.taskId === task.id)?.score || 0;
+        updateTaskScore(task.id, currentScore + 1);
+      }
+    });
+
+    // Move to next question or finish
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex(currentQuestionIndex + 1);
+    } else {
+      // Survey completed
+      Alert.alert(
+        'Journey Complete',
+        'Your personalized recommendations are ready.',
+        [
+          {
+            text: 'View Results',
+            onPress: () => navigation.navigate('Result' as never),
+          },
+        ]
+      );
+    }
+  };
+
+
+
+  const handleReset = () => {
+    Alert.alert(
+      'Start Fresh',
+      'Are you sure you want to begin your journey again?',
+      [
+        { text: 'Cancel', style: 'cancel' },
+        {
+          text: 'Begin Again',
+          style: 'destructive',
+          onPress: () => {
+            resetScores();
+            setCurrentQuestionIndex(0);
+          },
+        },
+      ]
+    );
+  };
+
+  if (questions.length === 0) {
+    return (
+      <View style={styles.container}>
+        <Text style={styles.noQuestionsText}>
+          No guides available. Please add some questions first.
+        </Text>
+      </View>
+    );
+  }
+
+  if (!currentQuestion) {
+    return (
+      <View style={styles.container}>
+        <Text style={styles.noQuestionsText}>No more questions.</Text>
```

Comment:
makes me think, if you could back in questionnaire, should it/does it show what you currently have selected? ... changing it would mean reversing the add 1 update

Reply:
can confirm, there is a bug here if you go back and keep adding yeses, it will add too many +1 to score